### PR TITLE
[BUG] Fixed the dimension-wise kl for batchTC 

### PR DESCRIPTION
**Pull request by linesd**
- Fixed dim-wise kl in losses.py.
- Added dim-wise kl to factor-vae so the loss can be estimated without the mutual information term
- Created new functions for the minibatch weighted and stratified sampling
- Removed old function to calculate dim-wise term which is no longer needed.
- Corrected a spelling error in hyperparam.ini.

### CODE REVIEW

File: hyperparam.ini [hunk 1]
```diff
@@ -37,7 +37,7 @@
     batchTC_A = 1
     batchTC_G = 1
     batchTC_B = 6
-    no_mss = False
+    no_mss = False #used for factor too
```
Comment (linesd) [inline hyperparam.ini]: Should this be a separate argument with the new cleaned up code?

**Comment by linesd** [comment]
Pull request now ready for review

File: disvae/models/losses.py [hunk 2]
```diff
@@ -235,8 +238,24 @@ def __call__(self, data, model, optimizer, storer):
             vae_loss = rec_loss + kl_loss + self.gamma * tc_loss
         else:
             # return vae loss without mutual information term
+            # change latent dist to torch.tensor (could probably avoid this)
+            latent_dist = torch.stack((latent_dist[0], latent_dist[1]), dim=2)
+            # calculate log p(z)
+            prior_params = torch.zeros(half_batch_size, latent_dist.size(1), 2)
+            logpz = log_density_normal(latent_sample1, prior_params, half_batch_size,
```
Comment (alecokas) [inline disvae/models/losses.py]: Maybe log_pz is more readable here? Just a personal preference

File: disvae/models/losses.py [hunk 3]
```diff
@@ -235,8 +238,24 @@ def __call__(self, data, model, optimizer, storer):
             vae_loss = rec_loss + kl_loss + self.gamma * tc_loss
         else:
             # return vae loss without mutual information term
+            # change latent dist to torch.tensor (could probably avoid this)
+            latent_dist = torch.stack((latent_dist[0], latent_dist[1]), dim=2)
+            # calculate log p(z)
+            prior_params = torch.zeros(half_batch_size, latent_dist.size(1), 2)
```
Comment (YannDubs) [inline disvae/models/losses.py]: that will break on GPU no ? .to(device)

File: hyperparam.ini (hunk 1 above)
Comment (YannDubs) [reply hyperparam.ini]: yes it should be one for both. But in any case CLI should not contain arguments that people will not change. I.e. i'll run both chose the best then remove the hyperparameter from CLI (although it's perfectly fine in the function)

File: disvae/models/losses.py (hunk 2 above)
Comment (linesd) [reply disvae/models/losses.py]: Going to leave as is for now.

File: disvae/models/losses.py (hunk 3 above)
Comment (linesd) [reply disvae/models/losses.py]: I have fixed this, now running on GPU as well.
